specs/docs/security/DataHubSecurity.md

```diff
@@ -0,0 +1,267 @@
+
+#### Example xdmp.securityAssert call
+```javascript
+xdmp.securityAssert("http://marklogic.com/data-hub/function-access", "execute");
+```
+
+### Granting Privileges To Function Via Amps
+
+Execute privileges for a given role should never be broader than needed for the operations a role needs to execute. Amps are a way to grant temporary roles inside a designated function. The function can then institute guardrails to ensure the roles are only used for the narrow intended purpose. See [Temporarily Increasing Privileges with Amps](https://docs.marklogic.com/guide/security/execute#id_54880).
+
+#### Example amp configuration
+```javascript
+{
+  "local-name" : "updateJob",
+  "document-uri" : "/data-hub/5/impl/jobs.sjs",
+  "modules-database" : "%%mlModulesDbName%%",
+  "role" : [ "data-hub-job-internal" ]
+}
+```
+
+In JavaScript, in addition to the amp configuration file, you need to make use of the `module.amp` function to wrap your function on export. See [Amps and the module.amp Function](https://docs.marklogic.com/guide/jsref/functions#id_13020) 
+
+#### Example JavaScript amp using module.amp
+```javascript
+module.exports.updateJob = module.amp(
+  function updateJob(datahub, jobId, status, flow, step, lastCompleted, stepResponse) {
+  //...
+  }
+);
+```
+### When To Grant Privileges To User Roles And When To Use Amps
+
+User roles should be assigned privileges when the privilege is sufficiently restrictive and does not allow arbitrary code execution.
+
+If the privilege would allow arbitrary code execution, that is a sure indication that an amp should be used. For other privileges, consider if the privilege can be used beyond the intended purpose. Where possible use granular privileges (See [Granular Privileges](https://docs.marklogic.com/guide/security/granular)). If the user role can't use granular privileges and the required privilege would be too broad, (1) a new narrow execute privilege should be created, (2) the function that requires the privilege should check for the narrow privilege, and (3) the function should be amped with the broader privilege.
+
+Roles to be used with amps should be suffixed with `-internal` to indicate that the role is used for amps and NOT intended to be assigned to users.
+
+## Data Hub Roles and Privileges
+
+### data-hub-common
+
+Data Hub role containing the common privileges across Data Hub roles.
+
+#### Inherited Roles
+ - data-hub-module-reader
+ - rest-reader
+
+#### Privileges
+
+Privilege | Notes
+---------|----------
+ xdmp-eval-in | Required to specify database for REST Client. (Note: This privilege alone doesn't allow eval to be called. The REST API code does the neccesary amping for eval execution)
+ xdbc-eval-in | Same as xdmp-eval-in
+
+### data-hub-common-writer
+
+Data Hub role containing the common privileges across Data Hub writer roles.
+
+#### Inherited Roles
+ - data-hub-common
+ - rest-writer
```

Comment:
We don't want to inherit this role, because it includes default permissions that work against document-level security. Need to inherit the rest-writer privilege instead.

Reply:
Changed to use the privilege.